cpp/src/command_classes/SwitchBinary.cpp

```diff
@@ -181,6 +221,60 @@ void SwitchBinary::SetValueBasic
 }
 
 //-----------------------------------------------------------------------------
+// <SwitchBinary::SetState>
+// Set a new state for the switch
+//-----------------------------------------------------------------------------
+bool SwitchBinary::SetState
+(
+	uint8 const _instance,
+	bool const _state
+)
+{
+	uint8 const nodeId = GetNodeId();
+	uint8 const targetValue = _state ? 0xff : 0;
+
+	Log::Write( LogLevel_Info, nodeId, "SwitchBinary::Set - Setting to %s", _state ? "On" : "Off");
+	Msg* msg = new Msg( "SwitchBinaryCmd_Set", nodeId, REQUEST, FUNC_ID_ZW_SEND_DATA, true );
+	msg->SetInstance( this, _instance );
+	msg->Append( nodeId );
+	
+	if( ValueByte* durationValue = static_cast<ValueByte*>( GetValue( _instance, SwitchBinaryIndex_Duration ) ) )
```

Comment:
I'd change this to a `GetVersion() =>2` test instead.

Reply:
https://github.com/OpenZWave/open-zwave/blob/Dev/cpp/src/command_classes/SwitchMultilevel.cpp#L477 should probably be changed aswell then.